src/server/pfs/server/driver.go

```diff
@@ -363,28 +362,24 @@ func (d *driver) deleteAllBranchesFromRepos(txnCtx *txncontext.TransactionContex
 }
 
 func (d *driver) deleteAllRepos(ctx context.Context) ([]*pfs.Repo, error) {
-	var repos, deleted []*pfs.Repo
-
+	var repoInfos []*pfs.RepoInfo
 	if err := d.listRepo(ctx, false, "", nil, func(repoInfo *pfs.RepoInfo) error {
-		repos = append(repos, repoInfo.Repo)
+		repoInfos = append(repoInfos, repoInfo)
 		return nil
 	}); err != nil {
 		return nil, err
 	}
-	if len(repos) == 0 {
+	if len(repoInfos) == 0 {
 		return nil, nil
 	}
-
+	var deleted []*pfs.Repo
 	if err := d.txnEnv.WithWriteContext(ctx, func(txnCtx *txncontext.TransactionContext) error {
-		// the list does not use the transaction
-		for _, repo := range repos {
-			if err := d.deleteRepo(txnCtx, repo, true); err != nil {
-				if errors.As(err, &auth.ErrNotAuthorized{}) {
-					continue
-				}
+		for _, ri := range repoInfos {
```

Comment:
The force approach seems kind of risky to me. I might not be against it for the time being, but we should consider doing a topological sort of the branches then order the deletes based on that.